### DISCUSSION TITLE
Side-by-Side: Erweiterung der HTML-Templates um die Aufrufparameter "leftLayer" und "rightLayer"

### DISCUSSION THREAD

**Discussion by cmairamhof** [e-netze/webgis-community#136] [Ideas]
Erweiterung  der HTML-Templates "WebGIS-TileServices-SideBySide" und "WebGIS-Service-SideBySide"
um die Aufrufparameter "leftLayer" und "rightLayer"; leftLayer und rightLayer könnten über eine Layer-ID oder über den Dienstnamen entsprechend Mapbuilder aufgerufen werden. 

derzeit z.B.
https://gis.ktn.gv.at/webgisviewer/App/PortalApp/atlas-mobile?app=Kartenvergleich&category=KAGIS-Projekte&center=13.984265693092095,46.64063569829442&scale=8858

**Comment by gis-eni** [comment]
Ich bau das für den nächsten Release einmal so ein:
(und falls du nicht warten willst, kannst du das ganz leicht selber nachziehen)

File ``portal/_templates/WebGIS-TileServices-SideBySide.html``:

bisher:

``` javascript
$("#layerswitch-left").webgis_basemapswitch('setServiceId', { serviceId: serviceIds[0] });
$("#layerswitch-right").webgis_basemapswitch('setServiceId', { serviceId: serviceIds[1] });
```

es werden also zum Begin immer der erste und der zweite Dienst angezeigt.
Diese beiden Zeilen kann durch folgendes Konstrukt erweitern ersetzen:

``` javascript
console.log(serviceIds);
const params = new URLSearchParams(window.location.search);
const leftLayerParameter = params.get('left-layer');
const rightLayerParameter = params.get('right-layer');

if (leftLayerParameter != rightLayerParameter
    && serviceIds.includes(leftLayerParameter)
    && serviceIds.includes(rightLayerParameter))
{
    $("#layerswitch-left").webgis_basemapswitch('setServiceId', { serviceId: leftLayerParameter });
    $("#layerswitch-right").webgis_basemapswitch('setServiceId', { serviceId: rightLayerParameter });
} else
{
    $("#layerswitch-left").webgis_basemapswitch('setServiceId', { serviceId: serviceIds[0] });
    $("#layerswitch-right").webgis_basemapswitch('setServiceId', { serviceId: serviceIds[1] });
}
```

Hier werden zuerst die Url Parameter ``layer-left`` und ``layer-right`` ausgelesen. Wenn beide übergeben, nicht gleich sind und in den Services enthalten, werden diese gesetzt. Sonst kommt einfach wieder der erste und zweite Dienst zum Zug.

Wie die Werte für die Parameter sein müssen, kannst du dir in der Console anschauen: ``console.log(serviceIds)``

Achtung:
Sobald du etwas in den HTML Templates änderst, solltest du die Datei umgenennen (auch das ``.mta`` File). Sonst sind die Änderungen beim Update weg. Hier ist das aber egal, wenn du das genau so übernimmst. Genau so steht es auch in nächsten Update drin.

**Reply by fanie-mak** [reply]
Die Url-Parameter sind "left-layer" und "right-layer".
